pkg/apis/policy/v1alpha1/types_certificaterequestpolicy.go

```diff
@@ -53,95 +53,96 @@ type CertificateRequestPolicyList struct {
 // CertificateRequestPolicySpec defines the desired state of
 // CertificateRequestPolicy.
 type CertificateRequestPolicySpec struct {
-	// Allowed is the set of attributes that are "allowed" by this policy. A
-	// CertificateRequest will only be considered permissible for this policy if
-	// the CertificateRequest has the same or less as what is allowed.  Empty or
-	// `nil` allowed fields mean CertificateRequests are not allowed to have that
-	// field present to be permissible.
+	// Allowed defines the attributes a CertificateRequest is allowed to have.
+	// A CertificateRequest can request _less_ than what is allowed,
+	// but _not more_, i.e. a CertificateRequest can request a subset of what
+	// is declared as allowed by the policy.
+	// Omitted fields declares that the equivalent CertificateRequest
+	// field _must_ be omitted or have an empty value for the request to be
+	// permitted.
 	// +optional
 	Allowed *CertificateRequestPolicyAllowed `json:"allowed,omitempty"`
 
-	// Constraints is the set of attributes that _must_ be satisfied by the
-	// CertificateRequest for the request to be permissible by the policy. Empty
-	// or `nil` constraint fields mean CertificateRequests satisfy that field
-	// with any value of their corresponding attribute.
+	// Constraints define fields that _must_ be satisfied by the
+	// CertificateRequest for the request to be allowed by this policy.
+	// Omitted fields will be satisfied by any value on the corresponding
+	// attribute on the request.
 	// +optional
 	Constraints *CertificateRequestPolicyConstraints `json:"constraints,omitempty"`
 
-	// Plugins define a set of plugins and their configuration that should be
-	// executed when this policy is evaluated against a CertificateRequest. A
-	// plugin must already be built within approver-policy for it to be
+	// Plugins define plugins and their configuration that should be
+	// executed when this policy is evaluated against a CertificateRequest.
+	// A plugin must already be built within approver-policy for it to be
 	// available.
```

Comment:
This isn't enough information for me to do something.

Is a plugin some code I had to compile? Is it something I had to define elsewhere? Where? Is it another CRD?

Reply:
Yes, a plugin is built into approver-policy at build-time. I have tried to explain this in my experimental plugin adding CEL validation support: https://github.com/erikgb/cel-approver-policy-plugin#installation

Reply:
I have tried to reword this now.